src/main/java/org/spongepowered/api/tag/Taggable.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.tag;
+
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.registry.DefaultedRegistryValue;
+
+/**
+ *
+ */
+public interface Taggable extends DataHolder, DefaultedRegistryValue {
```

Comment:
I feel like this needs utility methods to get the tags associated with this object. 

If a tag is going to have a generic, should this have the same generic?

Reply:
This will definitely need utility methods and due to that, a self-referencing generic.

Reply:
I've added the utility methods - I hid them in the implementation as they were to cluttery/confusing for `default` in my opinion. Furthermore we might want to change how we retrieve them, or cache them etc.